addons/web/static/tests/views/control_panel_tests.js

```diff
@@ -466,6 +466,37 @@ QUnit.module('Views', {
         controlPanel.destroy();
     });
 
+    QUnit.test('invisible fields and filters with unknown related fields should not be rendered', async function (assert) {
+        assert.expect(2);
+
+        // Emuate that for fields has a "groups" attribute set on the model;
+        // of which the user is not member of; in such case those fields are
+        // removed from `fields_get()` call but as it's not supported by
+        // MockServer, we need to emulate that behavior manually
+        delete this.data['partner'].fields['bar'];
+        delete this.data['partner'].fields['date_field'];
+
+        var controlPanel = await createControlPanel({
+            model: 'partner',
+            arch: "<search>" +
+                        "<field name=\"foo\"/>" +
+                        "<filter name=\"filterDate\" string=\"Date\" date=\"date_field\" default_period=\"last_365_days\" invisible=\"1\" groups=\"base.group_system\"/>" +
+                        "<filter name=\"groupByBar\" string=\"Bar\" context=\"{'group_by': 'bar'}\" invisible=\"1\" groups=\"base.group_system\"/>" +
+                    "</search>",
+            data: this.data,
+            context: {
+                search_disable_custom_filters: true,
```

Comment:
I don't think this is useful for the test

Reply:
removed